rust/main/agents/relayer/src/server/message_retry.rs

```diff
@@ -128,12 +134,41 @@ mod tests {
         }
     }
 
+    async fn send_retry_responses_task(
```

Comment:
nit: this is technically a future rather than a task, even after the `join` it's still a future that executes the joined futures concurrently